src/components/forms/validator.ts

```diff
@@ -89,13 +86,14 @@ export const minMaxLength = (minLen: number, maxLen: number) => (value: string):
 
 export const ADDRESS_REPEATED_ERROR = 'Address already introduced'
 
-export const uniqueAddress = (addresses: string[] | List<string>): GenericValidatorType =>
-  memoize(
-    (value: string): ValidatorReturnType => {
-      const addressAlreadyExists = addresses.some((address) => sameAddress(value, address))
-      return addressAlreadyExists ? ADDRESS_REPEATED_ERROR : undefined
-    },
-  )
+export const uniqueAddress = (addresses: string[]): GenericValidatorType => (): ValidatorReturnType => {
+  const uniqueAddresses = new Set(addresses)
+
+  if (uniqueAddresses.size !== addresses.length) {
+    return ADDRESS_REPEATED_ERROR
+  }
+  return undefined
+}
```

Comment:
This would allow pasting the same address as checksummed/non-checksummed ones. This is why we used `sameAddress` function in the previous version of the validator

Reply:
It's fixed now